File: src/content/cpp/factor.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int number;
    cout << "Enter a number: ";
    cin >> number;

    cout << "Prime factors of " << number << " are: ";
    while (number % 2 == 0) {
        cout << 2;
        number /= 2;
    }

    for (int i = 3; i * i <= number; i += 2) {
        while (number % i == 0) {
            cout << i << " ";
            number /= i;
        }
    }

    if (number > 2)
        cout << number;

    cout << endl;

    return 0;
}

File: src/content/dart/factor.dart
import 'dart:io';

void main() {
  stdout.write('Enter a number: ');
  int number = int.parse(stdin.readLineSync()!);

  stdout.write('Prime factors of $number are: ');

  for (int i = 2; i <= number; i++) {
    while (number % i == 0) {
      stdout.write('$i ');
      number ~/= i;
    }
  }

  stdout.writeln();
}
